### PR TITLE
Add Delete Message Functionality

### LINKED ISSUE #69: Delete Message Feature
As an EmbedChat developer

**I need to**: Add an option to delete a message if the message belongs to that user.

**So That**: EmbedChat Users can delete their own messages.

**Acceptance Criteria**

- [ ] Implement the Delete functionality and the messages should be updated in realtime.

### PR DESCRIPTION
**Pull request by ayush3160**
# Implemented delete message functionailty. 

## Acceptance Criteria fulfillment

- [x] Implement the Delete functionality and the messages should be updated in realtime.

Fixes #69

### CODE REVIEW

**Comment by ayush3160** [comment]
@sidmohanty11 , Can you please review this PR. Let me know if any changes are required.

Thanks!

**Comment by sidmohanty11** [comment]
@ayush3160 great work! Thanks for this PR!

File: src/lib/api.js [hunk 1]
```diff
@@ -170,6 +170,18 @@ export default class RocketChatInstance {
...
+      await this.rcClient.subscribeRoom(this.rid);
+      await this.rcClient.onStreamData('stream-notify-room',(ddpMessage) => {
+        const [roomId, event] = ddpMessage.fields.eventName.split('/');
+
+        if(roomId !== this.rid){
+          return;
+        }
+
+        if(event === 'deleteMessage'){
+          callback(ddpMessage)
```
Comment (sidmohanty11) [inline src/lib/api.js:182]: neat!

File: src/components/MessageList/MessageList.js [hunk 2]
```diff
@@ -71,6 +72,10 @@ const MessageList = ({ messages, handleGoBack }) => {
     }
   };
```
Comment (sidmohanty11) [inline src/components/MessageList/MessageList.js:74]: Just one suggestion, we could add a toast notification that "Message deleted successfully" and handle any errors there.

**Review by sidmohanty11 — CHANGES_REQUESTED** [review]

**Comment by ayush3160** [comment]
@sidmohanty11 , Please have a look.

File: src/components/MessageList/MessageList.js [hunk 3]
```diff
@@ -148,6 +167,18 @@ const MessageList = ({ messages, handleGoBack }) => {
                       icon="pin"
                       onClick={() => handlePinMessage(msg)}
                     />
+                    {
+                      msg.u._id === authenticatedUserId && (
+                        <ActionButton
```
Comment (sidmohanty11) [inline src/components/MessageList/MessageList.js:172]: Why do we need an action button? I think it should be messagetoolbox item to keep it semantically aligned.

**Review by sidmohanty11 — CHANGES_REQUESTED** [review]

File: src/components/MessageList/MessageList.js (hunk 3 above)
Comment (ayush3160) [reply src/components/MessageList/MessageList.js]: @sidmohanty11 , I have replaced it with the message toolbox item.

Thanks for the suggestion.

**Review by sidmohanty11 — APPROVED** [review]
LGTM! Thanks @ayush3160

**Pull request merged**